crawlers/mooncrawl/mooncrawl/stats_worker/queries.py

```diff
@@ -57,52 +72,71 @@ def data_generate(
     """
     s3 = boto3.client("s3")
 
-    with yield_db_read_only_session_ctx() as db_session:
+    # Create session
+    engine = create_moonstream_engine(
+        MOONSTREAM_DB_URI_READ_ONLY,
+        pool_size=1,
+        statement_timeout=MOONSTREAM_QUERY_API_DB_STATEMENT_TIMEOUT_MILLIS,
+    )
+    process_session = sessionmaker(bind=engine)
+    db_session = process_session()
+
+    try:
 
         try:
             db_session.execute("SELECT 1")

```

Comment:
We can add `pre_ping` since we creating engine here and write one `try:...catch:...` block

Reply:
With this version lost part with `finaly`
```
finally:
        db_session.close()
 ```

Reply:
fixed